ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible_collections.arista.avd.plugins.plugin_utils.schema.errors import AvdSchemaError, AvdValidationError, AristaAvdError
+from ansible_collections.arista.avd.plugins.plugin_utils.schema.avdvalidator import AVD_META_SCHEMA, AvdValidator
+from ansible_collections.arista.avd.plugins.plugin_utils.schema.avddataconverter import AvdDataConverter
+
+try:
+    import jsonschema
+except ImportError as imp_exc:
+    JSONSCHEMA_IMPORT_ERROR = imp_exc
+else:
+    JSONSCHEMA_IMPORT_ERROR = None
+
+try:
+    from deepmerge import always_merger
+except ImportError as imp_exc:
+    DEEPMERGE_IMPORT_ERROR = imp_exc
+else:
+    DEEPMERGE_IMPORT_ERROR = None
+
+DEFAULT_SCHEMA = {
+    "type": "dict",
+    "allow_other_keys": True,
+}
+
+
+class AvdSchema():
+    def __init__(self, schema: dict = None):
+        if JSONSCHEMA_IMPORT_ERROR:
+            raise AristaAvdError('Python library "jsonschema" must be installed to use this plugin') from JSONSCHEMA_IMPORT_ERROR
+        if DEEPMERGE_IMPORT_ERROR:
+            raise AristaAvdError('Python library "deepmerge" must be installed to use this plugin') from DEEPMERGE_IMPORT_ERROR
+
+        if not schema:
+            schema = DEFAULT_SCHEMA
+        self._schema_validator = jsonschema.Draft7Validator(AVD_META_SCHEMA)
+        self.load_schema(schema)
+
+    def validate_schema(self, schema: dict):
+        validation_errors = self._schema_validator.iter_errors(schema)
+        for validation_error in validation_errors:
+            yield self._errror_handler(validation_error)
+
+    def load_schema(self, schema: dict):
+        for validation_error in self.validate_schema(schema):
+            raise validation_error
+        self._schema = schema
+        try:
+            self._validator = AvdValidator(schema)
+            self._dataconverter = AvdDataConverter(schema)
+        except Exception as e:
+            raise AristaAvdError('An error occured during creation of the validator') from e
+
+    def extend_schema(self, schema: dict):
+        for validation_error in self.validate_schema(schema):
+            raise validation_error
+        always_merger.merge(self._schema, schema)
+        for validation_error in self.validate_schema(self._schema):
+            raise validation_error
+
+    def validate(self, data, schema: dict = None):
+        if schema:
+            for schema_validation_error in self.validate_schema(schema):
+                yield schema_validation_error
+                return
```

Comment:
it will not. Since we return a generator, for every iteration it will stop at the first return. It will not stop with a yield. So this is a way to avoid the code below (actual validation in this case) to happen if we meet a schema_validation_error.